COHHIO/ryha
# Add funding source filter

# Overview
Funder filter is added to the application.

The filter is added to the right sidebar, above Project filter.

Funder filter affects Project filter: when Funder(s) selection changes, all Projects from selected Funder(s) are selected. The reason behind this logic is as follows: a project can have one or more funders. It didn't make sense in my opinion to allow users to select projects that were not funded by selected funders. A popover was added to project filter label to provide these information to users.

# Details
Adding funder filter involved several changes, as the information was stored in a .csv not currently processed:

- `read_funder()` function was created
- HMISmetadata dataset was updated (to account for the necessity of the new .csv)
- A new "codes" dataset was added. In this case, we added all described options (even though it is expected that only a few will be present in the app)
- Filter workflow was modified, making project selection depend on funder filter. The rest of the logic remained unchanged (as we are making projects depend on funders, we don't need to include "funder filtering logic" to any other place in the code, as this is being taken care with the project filtering)
- SQL create table statement was added for funder table

These changes are meant to be released on new upload period (around January 2025), where we will clean the database and start from an empty database (to start capturing Funder information in the database). There are no plans to retroactively gather this data.

# Test
- Run postgres/utils/recreate_dev.R
- Launch the application in dev and upload a test file
- Relaunch the application (to update the dev database)
- Check that funder filter works accordingly

Closes #44
